lib/browser/property-descriptor.ts

```diff
@@ -67,18 +67,39 @@ function canPatchViaPropertyDescriptor() {
   // by default XMLHttpRequest.prototype.onreadystatechange is undefined
   // without adding enumerable and configurable will cause onreadystatechange
   // non-configurable
-  Object.defineProperty(XMLHttpRequest.prototype, 'onreadystatechange', {
-    enumerable: true,
-    configurable: true,
-    get: function() {
-      return true;
-    }
-  });
-  const req = new XMLHttpRequest();
-  const result = !!req.onreadystatechange;
-  // restore original desc
-  Object.defineProperty(XMLHttpRequest.prototype, 'onreadystatechange', xhrDesc || {});
-  return result;
+  // and if XMLHttpRequest.prototype.onreadystatechange is undefined,
+  // we should set a real desc instead a fake one
+  if (xhrDesc) {
```

Comment:
@DaveMBush , the function canPatchDescriptor will see whether zone.js can patch global object, if return false, zone.js will not patch DOM,XHR, so everything will not in zone. So we have to pick up a global object to try to patch and see the result.